### DISCUSSION TITLE
[New Feature] Allow looping sounds

### DISCUSSION THREAD

**Discussion by Crimsonum** [Vinifera-Developers/Vinifera#1314] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

Having the ability to loop sounds, like in Red Alert 2, would greatly benefit the game's immersion and ambience. Within the current limitations, a looping sound for things like waterfalls cannot be reliably produced (the sound can be attached to the waterfall animation, but it would only play once during each animation loop, making it rely on animation/game speed).

### Proposed Code (optional)

New sound.ini key for sound entries:
`Loops= (bool, def false) ; if set, the sound loops infinitely until the attached object is destroyed or deleted`

**Comment by CCHyper** [comment]
Related to #5.

The audio engine in Tiberian Sun is pretty much the same as what was used in Red Alert, so it inherits its limitations. Once we can get the audio engine replaced, we can look at adding support for audio controllers.
